iomgr/lib/dbengine/parser/DBEngineSqlRequestFactory.cpp

```diff
@@ -310,6 +313,32 @@ requests::DBEngineRequestPtr DBEngineSqlRequestFactory::createSelectRequestForSi
             std::move(orderBy), std::move(offset), std::move(limit));
 }
 
+requests::DBEngineRequestPtr DBEngineSqlRequestFactory::createSelectRequestForShowTablesStatement()
```

Comment:
@niolap 
Such implementation of SHOW TABLES is possible as "it is better than nothing", but the output will be a bit messy. I'd expect, for example, to have table names sorted. Such implementation will output them in the order tables were created and you can't add sorting by name easily - Siodb does not implement ORDER BY and indexes required for it to function efficiently.

Better implementation would not access table SYS_TABLES at all, instead it should use dictionary of tables in the database object to get all table names and descriptions, then sort them, and then output to client in a way similar to what SHOW DATABASES does. But this is a bit more coding. But significantly less than adding indexes and ORDER BY.

Anyway, good as starter implementation.
Please fix code style issues (see other comments), and I will merge this.

Reply:
@cxxman thank you 👍 Yes, it's a quick win because I often want to get the table list without typing the full query. 
I created a ticket for further improvement: #110